samcli/local/apigw/local_apigw_service.py

```diff
@@ -169,10 +170,16 @@ def _request_handler(self, **kwargs):
         Response object
         """
         route = self._get_current_route(request)
+        cors_headers = Cors.cors_to_headers(self.api.cors)
+
+        method, _ = self.get_request_methods_endpoints(request)
+        if method == 'OPTIONS':
+            headers = Headers(cors_headers)
+            return self.service_response('', headers, 200)
 
         try:
             event = self._construct_event(request, self.port, self.api.binary_media_types, self.api.stage_name,
-                                          self.api.stage_variables)
+                                          self.api.stage_variables, cors_headers)
```

Comment:
Is this right? Cors_headers always get added to the event?

Reply:
Yes. I believe CORS needs to be added to every request.